specification/cognitiveservices/data-plane/ContentSafety/stable/2024-09-01/contentsafety.json

```diff
@@ -0,0 +1,1304 @@
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "PagedTextBlocklistItem": {
+      "type": "object",
+      "description": "Paged collection of TextBlocklistItem items",
+      "properties": {
+        "value": {
+          "type": "array",
+          "description": "The TextBlocklistItem items on this page",
+          "items": {
+            "$ref": "#/definitions/TextBlocklistItem"
+          },
+          "x-ms-identifiers": []
+        },
+        "nextLink": {
+          "type": "string",
+          "format": "uri",
+          "description": "The link to the next page of items"
+        }
+      },
+      "required": [
+        "value"
+      ]
+    },
+    "PromptInjectionAnalysisResult": {
+      "type": "object",
+      "description": "The individual analysis result of potential direct or indirect injection attacks.",
+      "properties": {
+        "attackDetected": {
+          "type": "boolean",
+          "description": "Whether a potential injection attack is detected or not."
+        }
+      },
+      "required": [
+        "attackDetected"
+      ]
+    },
+    "RemoveTextBlocklistItemsOptions": {
+      "type": "object",
+      "description": "The request to remove blocklistItems from a text blocklist.",
+      "properties": {
+        "blocklistItemIds": {
+          "type": "array",
+          "description": "Array of blocklistItemIds to remove.",
+          "items": {
+            "type": "string"
+          }
+        }
+      },
+      "required": [
+        "blocklistItemIds"
+      ]
+    },
+    "ShieldPromptOptions": {
+      "type": "object",
+      "description": "The request of analyzing potential direct or indirect injection attacks.",
+      "properties": {
+        "userPrompt": {
+          "type": "string",
+          "description": "The user prompt needs to be analyzed, which may contain direct injection attacks.",
+          "maxLength": 10000
+        },
+        "documents": {
+          "type": "array",
+          "description": "The documents needs to be analyzed, which may contain direct or indirect injection attacks.",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "ShieldPromptResult": {
+      "type": "object",
+      "description": "The combined analysis results of potential direct or indirect injection attacks.",
+      "properties": {
+        "userPromptAnalysis": {
+          "$ref": "#/definitions/PromptInjectionAnalysisResult",
+          "description": "Direct injection attacks analysis result for the given user prompt."
+        },
+        "documentsAnalysis": {
+          "type": "array",
+          "description": "Direct and indirect injection attacks analysis result for the given documents.",
+          "items": {
+            "$ref": "#/definitions/PromptInjectionAnalysisResult"
+          },
+          "x-ms-identifiers": []
+        }
```

Comment:
You are using a single schema for userPromptAnalysis result and the documentsAnalysis result. That may be fine today, but if these diverge in the future you'll need to change one of them and that will be a breaking change. Best to make them different schemas now to allow for this change in a non-breaking way in the future.

Reply:
Separated into two.